File: c#/src/Docmost.Sdk/Models/UpdateSpaceDto.cs
namespace Docmost.Sdk.Models;

public class UpdateSpaceDto
{
    [System.Text.Json.Serialization.JsonPropertyName("spaceId")]
    public string SpaceId { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("name")]
    public string? Name { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("slug")]
    public string? Slug { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("description")]
    public string? Description { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("disablePublicSharing")]
    public bool? DisablePublicSharing { get; set; }

}

File: go/docmost/models/update_space_dto.go
package models

// UpdateSpaceDto from OpenAPI.
type UpdateSpaceDto struct {
	SpaceId string `json:"spaceId"`
	Name *string `json:"name,omitempty"`
	Slug *string `json:"slug,omitempty"`
	Description *string `json:"description,omitempty"`
	DisablePublicSharing *bool `json:"disablePublicSharing,omitempty"`
}
